homeassistant/components/media_player/yamaha.py

```diff
@@ -110,10 +149,16 @@ def __init__(self, name, receiver, source_ignore, source_names):
         self._name = name
         self._zone = receiver.zone
 
+    def __eq__(self, other):
```

Comment:
I'm gonna try to get going on this again... Doing what you say gives me "unhashable type" (when checking in the set), so I assume I either have to implement __hash__ or check unique_id for each item? But this got me thinking, shouldn't there be functionality provided for this use case in a more generic place? I mean, every time a platform implements a service that is not part of the base component, the service handler will need to get a handle to one or more entity objects in order to call the correct method. By just looking at some media players, I can see that sonos, kodi and soundtouch performs similar logic. Now yamaha will need it and I had to add it to the Apple TV platform as well. Not sure if this applies to other components, but it seems like media players use it quite extensively and maybe it would be a good idea to extend the base component somehow. Any thoughts?